internal/controllers/cryostat_controller_test.go

```diff
@@ -831,6 +836,67 @@ var _ = Describe("CryostatController", func() {
 				t.expectPVC(test.NewDefaultPVCWithLabel())
 			})
 		})
+		Context("with an existing PVC", func() {
```

Comment:
Just a thought: 

Should we refactor this test to assume exising PVC is valid, then we add two case:

1. Valid -> reconcile -> successfuly updates
2. Invalid -> Inject client -> reconcile -> Then, check recorder.

Just to be consistent flow between cases in this `existing PVC` context and closer to PVC is updated with our controller. What do you think?

Reply:
Perhaps I am misunderstanding, but I think what you're describing is what the tests currently do. These new test cases create a valid PVC as a precondition to the test. The success case checks whether this PVC is updated (where possible) to match the config in the CR. The failure case injects the client which rejects the PVC update and checks the recorder.

Reply:
Oh wait, right! Sorry, I was looking at the overall structure. But the better phrasing would be: for the second `invalid` case, should we add an existing PVC in default object list in `BeforeEach`?

Reply:
https://github.com/ebaron/cryostat-operator/blob/86b057dce4ad7439bd15fd15968a8d60cfa47f90/internal/controllers/cryostat_controller_test.go#L868

Reply:
Ah, I see what you mean. In the failure case, the existing PVC is created by the controller. I'll see if I can rework it a bit so the success and failure cases have a more similar flow.

Reply:
I've revised the tests, let me know what you think. While it appears simpler, one drawback is it relies on the ordering that the controller reconciles object types. PVCs are one of the first now, but if they got moved after the TLS setup for example, once call of Reconcile wouldn't be sufficient because we need to update Certificate statuses.

I have an idea to not rely on this `reconcileCryostatFully` to do mock status updates in a particular order. This causes flexibility problems when writing tests, like this one. I'll save that for another PR though.